src/NuGet.Core/NuGet.Protocol/RemoteRepositories/HttpFileSystemBasedFindPackageByIdResource.cs

```diff
@@ -598,7 +601,31 @@ private class PackageInfo
 
             public string Path { get; set; }
 
-            public string ContentUri { get; set; }
+            public string BaseUri { private get; set; }
+            public string Id { private get; set; }
+
+            public string ContentUri
+            {
+                get
+                {
+                    var normalizedVersionString = Identity.Version.ToNormalizedString();
```

Comment:
I'm guessing that ContentUri typically won't be used more than once. The dependency resolver will probably only get the property when it's about to make an HTTP request to the URL, and once it does that, it has no reason to ever request the URL again.

But I'm wondering if it's worthwhile caching in a field, just in case, so the second get is cheap?

Reply:
I don't have the numbers on the calling patterns on this still, but it felt like it was called once infrequently, and more than that almost never, so the overhead associated with another field seemed worse than just recalculating.